syscall.go

```diff
@@ -21,6 +21,11 @@ func SyscallN(fn uintptr, args ...uintptr) (r1, r2, err uintptr) {
 	if len(args) > maxArgs {
 		panic("too many arguments to SyscallN")
 	}
+	if len(args) <= 3 {
```

Comment:
* Could you write a comment why we needed this?
* Do functions with 4-6 arguments work as expected?

Reply:
I believe so but just in case I used the 6 version instead of the 3 version. The tests still seem to pass.